pkg/oc/cli/cmd/newapp.go

```diff
@@ -385,13 +390,45 @@ func (o *NewAppOptions) RunNewApp() error {
 			fmt.Fprintf(out, "%sTrack installation of %s with '%s logs %s'.\n", indent, installing[i].Name, o.BaseName, installing[i].Name)
 		}
 	case len(result.List.Items) > 0:
+		//if we dont find a route we give a message to expose it
+		if !containsRoute(result.List.Items) {
+			//we if dont have any route but have services - suggest commndas to expose those
+			svc := getAllServices(result.List.Items)
+			if len(svc) > 0 {
+				fmt.Fprintf(out, "%sApplication is not exposed. You can expose it to outside world by executing one or all of the commands below \n", indent)
+				for _, s := range svc {
+					fmt.Fprintf(out, "%s '%s %s svc/%s' \n", indent, o.BaseName, ExposeRecommendedName, s)
+				}
+			}
+		}
 		fmt.Fprintf(out, "%sRun '%s %s' to view your app.\n", indent, o.BaseName, StatusRecommendedName)
 	}
 	return nil
 }
 
 type LogsForObjectFunc func(object, options runtime.Object, timeout time.Duration) (*restclient.Request, error)
 
+func containsRoute(item []runtime.Object) bool {
+	for _, i := range item {
+		switch i.(type) {
+		case *routeapi.Route:
+			return true
+		}
+	}
+	return false
+}
+
+func getAllServices(item []runtime.Object) []string {
```

Comment:
`func getServices(items []runtime.Object) []*kapi.Service` would be a better signature